### PR TITLE
Consolidate x-plat integration tests

### PR DESCRIPTION
**Pull request by qmfrederik**
Use a build matrix to avoid repeating the same code.

### CODE REVIEW

**Comment by qmfrederik** [comment]
@AArnott The Windows failure looks unrelated, could you restart the job?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -183,54 +183,34 @@ stages:
...
+        Ubuntu_Bionic_2_1:
+          containerImage: 'bionic-2.1'
+        Ubuntu_Bionic:
+          containerImage: 'bionic'
+        Ubuntu_Focal:
+          containerImage: 'focal'
+        Ubuntu_Disco_3_0:
+          containerImage: 'disco-3.0'
+        Arch_Linux:
+          containerImage: 'disco-3.0'
```
Comment (AArnott) [inline azure-pipelines.yml:201]: copy paste error. this should be `archlinux`. This caused a failure in AzP.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -183,54 +183,34 @@ stages:
...
       displayName: Install git
+      condition: ne(variables['installGit'], '')
     - template: azure-pipelines/xplattest-pipeline.yml
 
   - job: macOS
     pool:
       vmImage: macOS-10.15
     steps:
-    - template: azure-pipelines/xplattest-pipeline.yml
+    - template: azure-pipelines/xplattest-pipeline.yml
```
Comment (AArnott) [inline azure-pipelines.yml:216]: nit: trailing EOL

File: azure-pipelines.yml [hunk 3]
```diff
@@ -183,54 +183,34 @@ stages:
...
-    container: disco-3.0
-    steps:
-    - template: azure-pipelines/xplattest-pipeline.yml
-
-  - job: Arch_Linux
+  - job: linux
+    strategy:
+      matrix:
+        xenial:
+          containerImage: 'xenial'
```
Comment (AArnott) [inline azure-pipelines.yml:190]: nit: remove unnecessary single-quotes

File: azure-pipelines.yml [hunk 4]
```diff
@@ -183,54 +183,34 @@ stages:
...
+          installGit: 'sudo apt update && sudo apt-get install -y git'
```
Comment (AArnott) [inline azure-pipelines.yml:191]: nit: `installGit` does more than just install git in some cases. Maybe call it `bashSetup`?

**Review by AArnott — CHANGES_REQUESTED** [review]
Thanks.

**Pull request merged**